ludwig/models/modules/image_encoders.py

```diff
@@ -187,36 +189,21 @@ def __init__(
             default_weights_regularizer=weights_regularizer,
             default_bias_regularizer=bias_regularizer,
             default_activity_regularizer=activity_regularizer,
-            # default_weights_constraint=weights_constraint,
-            # default_bias_constraint=bias_constraint,
+            # default_weights_constraint=fc_weights_constraint,
+            # default_bias_constraint=fc_bias_constraint,
             default_norm=norm,
             default_norm_params=norm_params,
             default_activation=activation,
             default_dropout_rate=dropout_rate,
         )
 
-    def __call__(
-            self,
-            input_image,
-            regularizer,
-            dropout,
-            is_training
-    ):
+    def call(self, inputs, training=None, mask=None):
+        inputs = tf.cast(inputs, tf.float32)
         # ================ Conv Layers ================
-        hidden = self.resnet(
-            input_image,
-            regularizer,
-            dropout,
-            is_training=is_training
-        )
-        hidden = self.flatten(hidden)
+        hidden = self.resnet(inputs, training)
+        hidden = tf.reshape(hidden, [hidden.shape[0], -1])
```

Comment:
there's a tf.keras.layers.Flatten if you prefer to use that, but it does the same thing really